src/components/coin.ts

```diff
@@ -105,9 +113,13 @@ export const adjustCoinBalanceByUserId = async (
   reason: string | null = null,
   adminId: string | null = null
 ): Promise<void> => {
-  const oldBalance = await getCoinBalanceByUserId(userId);
-  const newBalance = Math.max(oldBalance + amount, 0);
-  await changeDbCoinBalanceByUserId(userId, oldBalance, newBalance, event, reason, adminId);
+  try {
+    const oldBalance = await getCoinBalanceByUserId(userId);
+    const newBalance = Math.max(oldBalance! + amount, 0);
+    await changeDbCoinBalanceByUserId(userId, oldBalance!, newBalance, event, reason, adminId);
+  } catch (e) {
+    console.log(e);
```

Comment:
> In addition, in coin.ts, there appears to be error catching, in which the error is then logged or nothing is done at all. In these cases, do we also want the user to be aware that there is an error?

This is one of the few instances of what I meant by "the error is then logged."

Reply:
Oh, whoops, that was from my debugging, I'll remove the try catch blocks from these